.githooks/pre-commit

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+# http://redsymbol.net/articles/unofficial-bash-strict-mode/
+set -euo pipefail
+IFS=$'\n\t'
+
+if ! ./gradlew spotlessCheck; then
```

Comment:
By committing this file you leave a developer without any way to configure his own pre-commit hooks. I think this is a very bad idea.

Reply:
how do you suggest providing this kind of functionality then?  I feel if we don't provide it, 99% of people won't automatically run it before commit, get a failure in CI, have to recommit due to failures.

Reply:
We currently provide an example script and a suggestion in CONTRIBUTING to symlink it here. I think this is enough. If a developer frequently have to recommit due to formatting issues, they will remember to actually use this hook.

I don't think we have to force all developers to use exactly one pre-commit hook blessed by us. Especially if this is "for their benefit".

Reply:
A. I wonder, how many developers will be confused by this name similarity. IMO from this POV old location, `buildscripts` was less confusing. But let it be.